### PR TITLE
Adding colors to the IHC

### PR DESCRIPTION
**Pull request by cespenel**
## Description
Following [google group discussion](https://groups.google.com/forum/#!topic/scikit-image/yumvSsa7irc), I added the colors to the IHC gallery.

## Checklist
[It's fine to submit PRs which are a work in progress! But before they are merged, all PRs should provide:]
- [ ] Clean style in [the spirit of PEP8](https://www.python.org/dev/peps/pep-0008/)
- [ ] [Docstrings for all functions](https://github.com/numpy/numpy/blob/master/doc/example.py)
- [ ] Gallery example in `./doc/examples` (new features only)
- [ ] Unit tests

[For detailed information on these and other aspects see [scikit-image contribution guidelines](http://scikit-image.org/docs/dev/contribute.html)]

## References
[If this is a bug-fix or enhancement, it closes issue # ]
[If this is a new feature, it implements the following paper: ]

### CODE REVIEW

**Comment by stefanv** [comment]
Thanks, @cespenel!  If anyone uses this in their work, please review this PR.

File: doc/examples/color_exposure/plot_ihc_color_separation.py [hunk 1]
```diff
@@ -23,7 +23,14 @@
 
 from skimage import data
 from skimage.color import rgb2hed
+from matplotlib.colors import LinearSegmentedColormap
 
+# Create an artificial color close to the orginal one
+cmap_Hema = LinearSegmentedColormap.from_list('mycmap', ['white', 'navy'])
+cmap_DAB = LinearSegmentedColormap.from_list('mycmap', ['white',
+'saddlebrown'])
```
Comment (soupault) [inline doc/examples/color_exposure/plot_ihc_color_separation.py]: Please, follow PEP8 for the indentation here and below.

**Comment by soupault** [comment]
One remark from my side.
I'm not an expert, but this looks better and the code is ok. So, if anyone'd like to accept this, feel free to merge. :+1:

File: doc/examples/color_exposure/plot_ihc_color_separation.py [hunk 2]
```diff
@@ -35,13 +42,13 @@
 ax[0].imshow(ihc_rgb)
 ax[0].set_title("Original image")
 
-ax[1].imshow(ihc_hed[:, :, 0], cmap=plt.cm.gray)
+ax[1].imshow(ihc_hed[:, :, 0], cmap=cmap_Hema)
```
Comment (sciunto) [inline doc/examples/color_exposure/plot_ihc_color_separation.py]: or `cmap_hema`?

File: doc/examples/color_exposure/plot_ihc_color_separation.py [hunk 3]
```diff
@@ -35,13 +42,13 @@
...
 ax[1].set_title("Hematoxylin")
 
-ax[2].imshow(ihc_hed[:, :, 1], cmap=plt.cm.gray)
+ax[2].imshow(ihc_hed[:, :, 1], cmap=cmap_Eosin)
```
Comment (sciunto) [inline doc/examples/color_exposure/plot_ihc_color_separation.py]: or `cmap_eosin`? Looks like a mix between camelcase and underscore conventions.

**Comment by sciunto** [comment]
One remark also mon my side but I'm don't use this feature.

**Comment by cespenel** [comment]
Thank you, I cleaned it.

**Comment by emmanuelle** [comment]
Hum, on the one hand this PR looks fine to me and I suggest to merge it; on the other hand the example seems very very domain-specific. I wonder if such functions (like ``data.immunohistochemistry``, ``color.rgb2hed``) should not be moved to separate projects within the scikit-image organisation (we could have skimage-bio, skimage-tomo -- imported as skimbio and skimtomo :-)). 

What do you think? 

About this specific example, it was written by @spotter quite a while ago. @cespenel, do you know if a lot of people could be using these functions, or is it very specific to immunohistochemistry?

**Comment by cespenel** [comment]
@emmanuelle, I only used the module for histology and I'm not sure what other application it could have. I do agree that having a Scikit BioImage gallery (or something like that) for confocal images etc. would be very useful, at least in my domain :-) ...

**Comment by emmanuelle** [comment]
@cespenel thanks for your insights.

@scikit-image/core what do you think about projects that would be more domain-specific?

**Comment by emmanuelle** [comment]
@scikit-image/core this PR looks good and is an improvement so I'm merging it. We can have the discussion about domain-specific projects somewhere else.

**Pull request merged**